drivers/virtualbox/virtualbox.go

```diff
@@ -288,26 +289,37 @@ func (d *Driver) Create() error {
 		return err
 	}
 
-	// let VBoxService do nice magic automounting (when it's used)
-	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountPrefix", "/"); err != nil {
-		return err
-	}
-	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountDir", "/"); err != nil {
-		return err
-	}
+	var shareMountDir, shareDir, username string
 
-	var shareName, shareDir string // TODO configurable at some point
 	switch runtime.GOOS {
 	case "windows":
-		shareName = "c/Users"
-		shareDir = "c:\\Users"
+		username = os.Getenv("USERNAME")
+		shareDir = "c:\\Users\\" + username
+		shareMountDir = "/c/Users"
 	case "darwin":
-		shareName = "Users"
-		shareDir = "/Users"
-		// TODO "linux"
+		username = os.Getenv("USER")
+		shareDir = path.Join("/Users", username)
+		shareMountDir = "/Users"
+	case "linux":
+		username = os.Getenv("USER")
+		shareMountDir = "/home"
+		shareDir = path.Join("/home", username)
+	}
+
+	shareName := username
+
+	log.Debugf("creating share: path=%s", shareDir)
+
+	// let VBoxService do nice magic automounting (when it's used)
+	if err := vbm("guestproperty", "set", d.MachineName, "/VirtualBox/GuestAdd/SharedFolders/MountDir", shareMountDir); err != nil {
```

Comment:
Perhaps, but don't they both get added together for the final result?  It still feels wrong to be setting this to `/home` or `/Users` since it's at the VM level not the share level, but I'm not sure what exactly the implications are since the docs for both of these are practically nonexistent. :disappointed: